src/flint/test/test.py

```diff
@@ -1972,17 +2356,21 @@ def setbad(obj, i, val):
         else:
             assert raises(lambda: P([2, 2]) / 2, TypeError)
 
-        assert raises(lambda: 1 / P([1, 1]), TypeError)
-        assert raises(lambda: P([1, 2, 1]) / P([1, 1]), TypeError)
-        assert raises(lambda: P([1, 2, 1]) / P([1, 2]), TypeError)
+        # TODO:
+        # I think this should be a ValueError and not a Type Error?
+        # assert raises(lambda: 1 / P([1, 1]), TypeError)
+        # assert raises(lambda: P([1, 2, 1]) / P([1, 1]), TypeError)
+        # assert raises(lambda: P([1, 2, 1]) / P([1, 2]), TypeError)
```

Comment:
The way this is defined for the other poly types is just that it is a TypeError. Note that `P([1, 2, 1]) / P([1, 1])` can be an exact division but it gives a TypeError: you need to use `//` or `divmod` instead. I think that this was a deliberate choice by @fredrik-johansson so that the operations are always well defined.

I did add scalar division where the ground domain is a field though (nmod_poly and fmpq_poly):
```python
In [14]: flint.fmpq_poly([1, 1]) / 7
Out[14]: 1/7*x + 1/7
```
In SymPy's algebra system there is another operation which is call exquo that performs an exact division when possible or otherwise raises an ExactQuotientFailed error. The `exquo` operation is very useful but it is not implemented as `a / b` in SymPy at least. You have to use it explicitly:
```python
In [16]: ZZ.exquo(ZZ(6), ZZ(3))
Out[16]: mpz(2)

In [17]: ZZ.exquo(ZZ(6), ZZ(5))
---------------------------------------------------------------------------
ExactQuotientFailed
```
Note also that `fmpz` refuses an exact division:
```python
In [18]: flint.fmpz(5)/5
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[18], line 1
----> 1 flint.fmpz(5)/5

TypeError: unsupported operand type(s) for /: 'flint.types.fmpz.fmpz' and 'int'
```

Reply:
Oh ok. Thank you for the detailed reply. Seems like the best thing then is to remove the support for division in this way and potentially have a specialised formula. 

For scalar division, this is a TODO atm but could easily be implemented to allow the work to match other polynomials.

Reply:
I don't know whether we would potentially want to change `a / b` to be something like `exquo`. There is a lot of inconsistency in defining `a / b` for non-fields between different environments even just within the Python ecosystem. Maybe it is better just to avoid that ambiguity by not allowing it at all.